Decred Wallet/Features/TransactionDetails/TransactionDetailsViewController.swift

```diff
@@ -8,262 +8,308 @@ import UIKit
+
+            let voteBits = TransactionDetail(
+                title: LocalizedStrings.voteBits,
+                value: self.transaction.voteBits,
+                isCopyEnabled: false
             )
             generalTxDetails.append(voteBits)
         }
     }
     
-    @objc func back() {
-        self.navigationController?.popViewController(animated: true)
-    }
-    
-    @objc func showOptionsMenu(sender: UIBarButtonItem) {
-        let alertController = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
-        
-        let cancelAction = UIAlertAction(title: LocalizedStrings.cancel, style: .cancel, handler: nil)
-        
-        let copyTxHash = UIAlertAction(title: LocalizedStrings.copyTransactionHash, style: .default, handler: { (alert: UIAlertAction!) -> Void in
-            self.copyText(self.transaction.hash)
-        })
-        
-        let copyRawTx = UIAlertAction(title: LocalizedStrings.copyRawTransaction, style: .default, handler: { (alert: UIAlertAction!) -> Void in
-            self.copyText(self.transaction.hex)
-        })
-        
-        let viewOnDcrdata = UIAlertAction(title: LocalizedStrings.viewOnDcrdata, style: .default, handler: { (alert: UIAlertAction!) -> Void in
-             if BuildConfig.IsTestNet {
-                self.openLink(urlString: "https://testnet.dcrdata.org/tx/\(self.transaction.hash)")
-             } else {
-                self.openLink(urlString: "https://dcrdata.decred.org/tx/\(self.transaction.hash)")
+    private func displayTitle() {
+        if self.transaction.type == DcrlibwalletTxTypeRegular {
+            if self.transaction.direction == DcrlibwalletTxDirectionSent {
+                self.txTypeLabel.text = LocalizedStrings.sent
+            } else if self.transaction.direction == DcrlibwalletTxDirectionReceived {
+                self.txTypeLabel.text = LocalizedStrings.received
+            } else if self.transaction.direction == DcrlibwalletTxDirectionTransferred {
+                self.txTypeLabel.text = LocalizedStrings.transferred
             }
-        })
-        
-        alertController.addAction(cancelAction)
-        alertController.addAction(copyTxHash)
-        alertController.addAction(copyRawTx)
-        alertController.addAction(viewOnDcrdata)
-        
-        if let popoverPresentationController = alertController.popoverPresentationController {
-            popoverPresentationController.barButtonItem = self.navigationItem.rightBarButtonItems![0]
+        } else if self.transaction.type == DcrlibwalletTxTypeVote {
+            self.txTypeLabel.text = LocalizedStrings.voted
+        } else if self.transaction.type == DcrlibwalletTxTypeTicketPurchase {
+            self.txTypeLabel.text = LocalizedStrings.ticket
         }
-        
-        self.present(alertController, animated: true, completion: nil)
+    }
 
+    @IBAction func onClose(_ sender: Any) {
+        self.dismiss(animated: true, completion: nil)
```

Comment:
Use `self.dismissView()` to better handle any potential scenario where this VC was pushed instead of presented.